spring-cloud-gcp-starters/spring-cloud-gcp-starter-logging/src/main/java/org/springframework/cloud/gcp/logging/autoconfig/StackdriverLoggingProperties.java

```diff
@@ -0,0 +1,35 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.logging.autoconfig;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+/**
+ * @author Chengyuan Zhao
+ */
+@ConfigurationProperties("spring.cloud.gcp.logging.traceId")
```

Comment:
The last concern: why do we drill so deep?
Why can't we just live with the `spring.cloud.gcp.logging` namespace and use `spring.cloud.gcp.logging.extractor` for the property? I guess we are not going to have some other configuration properties for the `traceId`

Reply:
I was thinking "extractor" is a pretty vague word and we might want to extract other things in the future (that aren't the trace ID).

Reply:
Good catch -- this should be just `spring.cloud.gcp.logging`.

If we want a `trace-id` element, we should add a nested property of type `TraceId`.

Reply:
Oh wow yea good point! Is there a way to specify just spring.cloud.gcp.logging for this whole class but specify traceId.extractor for just that class member?

Reply:
Yes, there is. Just call the property as `traceIdExtractor`. That's the best I can suggest.